dtfs-central-api/src/helpers/get-keying-sheet-calculation-tfm-facility-values.ts

```diff
@@ -4,34 +4,37 @@ import { TfmFacilitiesRepo } from '../repositories/tfm-facilities-repo';
 import { NotFoundError } from '../errors';
 import { convertTimestampToDate } from './convert-timestamp-to-date';
 import { getEffectiveCoverEndDateAmendment } from './amendments/get-effective-cover-end-date-amendment';
+import { KeyingSheetCalculationTfmFacilityValues } from '../types/tfm/tfm-facility';
 
 /**
- * Gets the latest values for the TFM facility with the supplied facility id
+ * Gets TFM facility values with the supplied facility id for keying sheet calculations
+ * if a report period is supplied, the values returned will take into account any amendments that came into effect before the end of the report period passed in.
+ * else will return the values as they were at facility creation ignoring all amendments
  * @param facilityId - The facility id
  * @param reportPeriod - The report period
- * @returns The latest values
+ * @returns TFM facility values - coverEndDate, coverStartDate, dayCountBasis, interestPercentage, coverPercentage, value
  */
-export const getLatestTfmFacilityValues = async (
+export const getKeyingSheetCalculationTfmFacilityValues = async (
   facilityId: string,
-  reportPeriod: ReportPeriod,
-): Promise<{
-  coverEndDate: Date;
-  coverStartDate: Date;
-  dayCountBasis: number;
-  interestPercentage: number;
-  coverPercentage: number;
-}> => {
+  reportPeriod?: ReportPeriod,
+): Promise<KeyingSheetCalculationTfmFacilityValues> => {
```

Comment:
I am not familiar with the flow/UX etc - wIll the keying sheet calculation ever _not_ be in TFM? Even if not a "TFM facility" surely the provided data would/should be the same?

I'm wondering if some namings could be simpler e.g

```
getKeyingSheetCalculationTfmFacilityValues
KeyingSheetCalculationTfmFacilityValues

// to
getKeyingSheetCalculationFacilityValues
KeyingSheetCalculationFacilityValues
```

Reply:
changed